sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -763,7 +751,7 @@ public void onSuccess(StorageObject response, HttpHeaders httpHeaders)
           @Override
           public void onFailure(GoogleJsonError e, HttpHeaders httpHeaders) throws IOException {
             IOException ioException;
-            if (errorExtractor.itemNotFound(e)) {
+            if (e.getCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
```

Comment:
`itemNotFound(e)` does a recursive check to find the exception, we could have a `STATUS_CODE_NOT_FOUND` in the cause but not the root exception, Can we retain usage of `errorExtractor.itemNotFound()` ?

Reply:
This is not possible anymore because ApiErrorExtrator doesn't support GoogleJsonError anymore by https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/327. I copied the same routine from the old ApiErrorExtrator to do the same thing.

Reply:
So this is a regression ? Note that Beam file IO connectors are very sensitives to changes in behavior of rename/copy/delete etc. since current behavior is carefully implemented (after many bugs) to be correct when there are step failures and retries.

Reply:
(unresloving)

Reply:
Yeah, `itemNotFound` method was removed by the recent gcsio so I copied the actual implementation of the method to keep it consistent.